core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -113,10 +159,14 @@ public void writeMessage(InputStream message, int length, @Nullable Runnable acc
       framer.writePayload(message, length);
```

Comment:
So if the stream is closed, we will end up returning false. We may want that to be true to drain any application buffers, but it could go either way. We would want NoopClientStream to match though.